File: sample-code/python/test/test_android_selectors.py
import pytest
import os
import copy

from appium import webdriver
from helpers import report_to_sauce, take_screenshot_and_logcat, ANDROID_BASE_CAPS, EXECUTOR


class TestAndroidSelectors():

    @pytest.fixture(scope='function')
    def driver(self, request, device_logger):
        calling_request = request._pyfuncitem.name

        caps = copy.copy(ANDROID_BASE_CAPS)
        caps['name'] = calling_request
        driver = webdriver.Remote(
            command_executor=EXECUTOR,
            desired_capabilities=caps
        )

        def fin():
            report_to_sauce(driver.session_id)
            take_screenshot_and_logcat(driver, device_logger, calling_request)
            driver.quit()

        request.addfinalizer(fin)

        driver.implicitly_wait(10)
        return driver

    def test_should_find_elements_by_accessibility_id(self, driver):
        search_parameters_element = driver.find_elements_by_accessibility_id('Content')
        assert 1 == len(search_parameters_element)

    def test_should_find_elements_by_id(self, driver):
        action_bar_container_elements = driver.find_elements_by_id('android:id/action_bar_container')
        assert 1 == len(action_bar_container_elements)

    def test_should_find_elements_by_class_name(self, driver):
        action_bar_container_elements = driver.find_elements_by_class_name('android.widget.FrameLayout')
        assert 3 == len(action_bar_container_elements)

    def test_should_find_elements_by_xpath(self, driver):
        action_bar_container_elements = driver.find_elements_by_xpath('//*[@class="android.widget.FrameLayout"]')
        assert 3 == len(action_bar_container_elements)

File: sample-code/csharp/test/TestAndroidSelectors.cs
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Appium;
using OpenQA.Selenium.Appium.Enums;
using OpenQA.Selenium.Appium.Android;
using OpenQA.Selenium.Remote;
using System;
using System.Collections.Generic;
using AppiumDotNetSamples.Helper;

namespace AppiumDotNetSamples
{
    [TestFixture()]
    public class AndroidSelectorsTest
    {
        private AndroidDriver<AndroidElement> driver;

        [OneTimeSetUp()]
        public void BeforeAll()
        {
            DesiredCapabilities capabilities = new DesiredCapabilities();
            capabilities.SetCapability(MobileCapabilityType.BrowserName, "");
            capabilities.SetCapability(MobileCapabilityType.PlatformName, App.AndroidDeviceName());
            capabilities.SetCapability(MobileCapabilityType.PlatformVersion, App.AndroidPlatformVersion());
            capabilities.SetCapability(MobileCapabilityType.AutomationName, "UIAutomator2");
            capabilities.SetCapability(MobileCapabilityType.DeviceName, "Nexus");

            capabilities.SetCapability(MobileCapabilityType.App, App.AndroidApp());

            driver = new AndroidDriver<AndroidElement>(Env.ServerUri(), capabilities, Env.INIT_TIMEOUT_SEC);
            driver.Manage().Timeouts().ImplicitWait = Env.IMPLICIT_TIMEOUT_SEC;
        }

        [OneTimeTearDown()]
        public void AfterAll()
        {
            driver.Quit();
        }

        [Test()]
        public void TestShouldFindElementsByAccessibilityId()
        {
            ICollection<AndroidElement> elements = driver.FindElementsByAccessibilityId("Content");
            Assert.AreEqual(1, elements.Count);
        }

        [Test()]
        public void TestShouldFindElementsById()
        {
            ICollection<AndroidElement> elements = driver.FindElementsById("android:id/action_bar_container");
            Assert.AreEqual(1, elements.Count);
        }

        [Test()]
        public void TestShouldFindElementsByClassName()
        {
            ICollection<AndroidElement> elements = driver.FindElementsByClassName("android.widget.FrameLayout");
            Assert.AreEqual(3, elements.Count);
        }

        [Test()]
        public void TestShouldFindElementsByXPath()
        {
            ICollection<AndroidElement> elements = driver.FindElementsByXPath("//*[@class='android.widget.FrameLayout']");
            Assert.AreEqual(3, elements.Count);
        }
    }
}
